File: src/main/java/ejerciciosJava/Ejercicio8.java
package ejerciciosJava;

import java.util.Scanner;

public class Ejercicio8 {
    public static void main() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Introduce la cadena para contar las vocales: ");
        String cadena = scanner.nextLine();
        int nA = 0, nE = 0, nI = 0, nO = 0, nU = 0;

        for (char caracter : cadena.toCharArray()) {
            switch (Character.toLowerCase(caracter)) {
                case 'a':
                    nA++;
                    break;
                case 'e':
                    nE++;
                    break;
                case 'i':
                    nI++;
                    break;
                case 'o':
                    nO++;
                    break;
                case 'u':
                    nU++;
                    break;
            }
        }
        System.out.println("A: " + nA);
        System.out.println("E: " + nE);
        System.out.println("I: " + nI);
        System.out.println("O: " + nO);
        System.out.println("U: " + nU);
    }
}

File: src/main/java/ejerciciosKotlin/Ejercicio8.kt
package ejerciciosKotlin

fun ejercicio8(){
    println("Introduce la cadena para contar las vocales: ")
    val cadena = readln()
    var nA = 0
    var nE = 0
    var nI = 0
    var nO = 0
    var nU = 0
    for (caracter in cadena){
        when(caracter.toString().lowercase()){
            "a" -> nA++
            "e" -> nE++
            "i" -> nI++
            "o" -> nO++
            "u" -> nU++
        }
    }
    println("A: $nA")
    println("E: $nE")
    println("I: $nI")
    println("O: $nO")
    println("U: $nU")
}